Source/ACE/Factories/GenericObjectFactory.cs

```diff
@@ -29,37 +26,14 @@ public static List<WorldObject> CreateWorldObjects(List<AceObject> sourceObjects
                     continue;
                 }
 
-                if ((oDescFlag & ObjectDescriptionFlag.LifeStone) != 0)
-                {
-                    results.Add(new Lifestone(aceO));
-                    continue;
-                }
-                if ((oDescFlag & ObjectDescriptionFlag.Portal) != 0)
+                if (aceO.Location != null)
                 {
-                   results.Add(new Portal(aceO));
-                   continue;
-                }
-                if ((oDescFlag & ObjectDescriptionFlag.Door) != 0)
-                {
-                    results.Add(new Door(aceO));
-                    continue;
-                }
+                    WorldObject wo = new GenericObject(aceO);
 
-                switch (oType)
-                {
-#if DEBUG
-                    default:
-                        // Use the DebugObject to assist in building proper objects for weenies
-                        // FIXME(ddevec): Some objects have null location.  This freaks out the landblock... ignore them?
-                        if (aceO.Location != null)
-                        {
-                            results.Add(new DebugObject(aceO));
-                        }
-                        break;
-#endif
+                    results.Add(wo.GetObjectFromWeenieType());
```

Comment:
Maybe I can play around with GenericObjectFactory itself and make that thing more useful there.. lemme look into that